WordPressKitTests/Utilities/MultipartFormTests.swift

```diff
@@ -0,0 +1,162 @@
+import Foundation
+import Alamofire
+import XCTest
+import CryptoKit
+
+@testable import WordPressKit
+
+class MutliparFormDataTests: XCTestCase {
+    struct Form: Codable {
+        struct Field: Codable {
+            var name: String
+            var content: String
+        }
+
+        var fields: [Field]
+
+        static func random(numberOfFields: Int = 10) -> Form {
+            let randomText: () -> String = { String(format: "%08x", Int.random(in: Int.min...Int.max)) }
+            let fields = (1...numberOfFields).map { _ in
+                Field(name: randomText(), content: randomText())
+            }
+            return Form(fields: fields)
+        }
+
+        func formDataUsingAlamofire() throws -> Data {
+            let formData = MultipartFormData()
+            formData.boundary = "testboundary"
+            for field in fields {
+                formData.append(field.content.data(using: .utf8)!, withName: field.name)
+            }
+            return try formData.encode()
+        }
+
+        func formData() throws -> Data {
+            try fields
+                .map {
+                    MultipartFormField(text: $0.content, name: $0.name)
+                }
+                .multipartFormDataStream(boundary: "testboundary")
+                .readToEnd()
+        }
+    }
+
+    func testRandomForm() throws {
+        let tempDir = FileManager.default.temporaryDirectory
+        let testData = tempDir.appendingPathComponent("test-form.json")
+        let afOutput = tempDir.appendingPathComponent("test-form.af.txt")
+        let wpOutput = tempDir.appendingPathComponent("test-form.wp.txt")
+
+        let form = Form.random()
+        try JSONEncoder().encode(form).write(to: testData)
+
+        let afEncoded = try form.formDataUsingAlamofire()
+        try afEncoded.write(to: afOutput)
+
+        let encoded = try form.formData()
+        try encoded.write(to: wpOutput)
+
+        add(XCTAttachment(contentsOfFile: testData))
+        add(XCTAttachment(contentsOfFile: afOutput))
+        add(XCTAttachment(contentsOfFile: wpOutput))
+
+        XCTAssertEqual(afEncoded, encoded)
+    }
+
+
+    func testPlainText() throws {
+        let af = MultipartFormData()
+        af.append("hello".data(using: .utf8)!, withName: "world")
+        af.append("foo".data(using: .utf8)!, withName: "bar")
+        af.append("the".data(using: .utf8)!, withName: "end")
+        let afEncoded = try af.encode()
+
+        print("# Alamofire")
+        try print(XCTUnwrap(String(data: afEncoded, encoding: .utf8)))
```

Comment:
I suppose we can remove these `print` statements before merging, right?

```suggestion
```

Reply:
Good call. They are gone in https://github.com/wordpress-mobile/WordPressKit-iOS/pull/654/commits/cab9ddf1046d83aa3d2045a89279f9ff6e352cc4.